src/test/ui/resolve/resolve-self-in-impl.rs

```diff
@@ -12,9 +12,13 @@ impl Tr for S where S<Self>: Copy {} // OK
 impl Tr for S where Self::A: Copy {} // OK
 
 impl Tr for Self {} //~ ERROR cycle detected
+//~^ ERROR `Self` is only available in impls, traits, and type definitions
```

Comment:
We should emit a single diagnostic for these, not multiple. I could see

> `Self` is only available in impls, traits, and type definitions

being a “help”, but I think it being a singular error makes more sense.

Reply:
Ok. I'll fix it。